### DISCUSSION TITLE
Architectural guidelines for developers

### DISCUSSION THREAD

**Discussion by febus982** [cloudevents/sdk-python#225] [General]
As mentioned in the other discussion (https://github.com/cloudevents/sdk-python/discussions/217) there are some packaging issues and legacy code that make the development experience complex and the result inefficient (especially impactful where lot of events might need to be processed).

Is the roadmap for V2 yet to be defined? If the answer is yes I'd like to contribute. Can I just send a proposal? Where?

If we document how V2 should look like, we could formalise the architectural documentation and the community could help in writing V2.

**Comment by xSAVIKx** [comment]
Hey, sorry, no work done in that direction yet.  I'd be glad to review the proposal. You can just create a PR with a `v2-roadmap.md` file and we can check it out in the PR if that works for you.

**Reply by febus982** [reply]
Cool, I'll prepare something in the next days. I have an idea in mind but I need some time to think about it and produce an acceptable document.
